feature/security/gnsi/authz/tests/authz_test.go

```diff
@@ -0,0 +1,752 @@
+	}
+	return user.spiffeID
+}
+
+func getTlsConfig(t *testing.T, dut *ondatra.DUTDevice, certName string) *tls.Config {
+	user, ok := usersMap[certName]
+	if !ok {
+		t.Fatalf("Could not find Spiffe ID for user %s", certName)
+	}
+	return user.tslConf
+}
+
+// Authz-1, Test policy behaviors, and probe results matches actual client results.
+func TestAuthz1(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	setUpUsers(t, dut)
+	tlsCertAdmin := getTlsConfig(t, dut, "cert_user_admin")
+	spiffeCertAdmin := getSpiffeID(t, dut, "cert_user_admin")
+	t.Run("Authz-1.1, - Test empty source", func(t *testing.T) {
+		// Pre-Test Section
+		_, policyBefore := authz.Get(t, dut)
+		t.Logf("Authz Policy of the Device %s before the Rotate Trigger is %s", dut.Name(), policyBefore.PrettyPrint())
+		defer policyBefore.Rotate(t, dut, uint64(time.Now().UnixMilli()), fmt.Sprintf("v0.%v", (time.Now().UnixNano())), false)
+
+		// Fetch the Desired Authorization Policy and Attach base Admin Policy Before Rotate
+		policies := loadPolicyFromJsonFile(t, dut, "testdata/policy.json")
+		newpolicy := getPolicyByName(t, "policy-everyone-can-gnmi-not-gribi", policies)
+		newpolicy.AddAllowRules("base", []string{*testInfraID}, []*gnxi.RPC{gnxi.RPCs.ALL})
+		// Rotate the policy.
+		newpolicy.Rotate(t, dut, uint64(100), "policy-everyone-can-gnmi-not-gribi_v1", false)
+
+		// Verification of Policy for cert_user_admin is allowed gNMI Get and denied gRIBI Get
+		t.Run("Verification of Policy for cert_user_admin is allowed gNMI Get and denied gRIBI Get", func(t *testing.T) {
+			authz.Verify(t, dut, spiffeCertAdmin, gnxi.RPCs.GRIBI_GET, tlsCertAdmin, true, true)
+			authz.Verify(t, dut, spiffeCertAdmin, gnxi.RPCs.GNMI_GET, tlsCertAdmin, false, true)
+		})
+
+	})
+
+	t.Run("Authz-1.2, Test Empty Request", func(t *testing.T) {
+		// Pre-Test Section
+		_, policyBefore := authz.Get(t, dut)
+		t.Logf("Authz Policy of the Device %s before the Rotate Trigger is %s", dut.Name(), policyBefore.PrettyPrint())
+		defer policyBefore.Rotate(t, dut, uint64(time.Now().UnixMilli()), fmt.Sprintf("v0.%v", (time.Now().UnixNano())), false)
+
+		// Fetch the Desired Authorization Policy and Attach base Admin Policy Before Rotate
+		policies := loadPolicyFromJsonFile(t, dut, "testdata/policy.json")
+		newpolicy := getPolicyByName(t, "policy-everyone-can-gribi-not-gnmi", policies)
+		newpolicy.AddAllowRules("base", []string{*testInfraID}, []*gnxi.RPC{gnxi.RPCs.ALL})
+		// Rotate the policy.
+		newpolicy.Rotate(t, dut, uint64(100), "policy-everyone-can-gribi-not-gnmi_v1", false)
+
+		// ensure   `cert_user_fake` is denied to issue `gNMI.Get` method.
+		// ensure   `cert_user_fake` is denied to issue `gRIBI.Get` method.
+		// ensure  `cert_user_admin` is allowed to issue `gRIBI.Get` method.
+		// ensure `cert_user_admin` is denied to issue `gNMI.Get` method.
+		t.Run("Verification of cert_user_fake and cert_user_admin premissions", func(t *testing.T) {
+			if false {
```

Comment:
Clarified in the README comments. 

Also, per the rule, fake_user should be allowed, right?

Reply:
soft verification yes (prob for user), hard verfication no, since the mtls connect will fail at certificate validation.

Reply:
The intent was not about validate certification error. It was about an unknown user.  Therefore, both soft and hard validation should pass.